### ISSUE TITLE
Rule request: Warn whenever any default value is provided in __post_init__

### ISSUE THREAD

**Issue by NeilGirdhar** [astral-sh/ruff#13128]
Default values should not be given in the arguments of `__post_init__`.  They should only be provided as default values for class-level _init-only_ member variables.

```python
from dataclasses import InitVar, dataclass


@dataclass
class C:
    x: InitVar[int] = 1

    def __post_init__(self, x: int = 4) -> None:
        print(x)  # noqa: T201


c = C()  # Prints 1!  4 is ignored.
```
The above text could be used for the Ruff rule warning message.

**Comment by dhruvmanila** [comment]
Interesting, thanks for raising this rule request. This seems like a reasonable rule to implement, most likely in a `RUF` category. What do you think? @AlexWaygood

**Comment by dhruvmanila** [comment]
Although, I couldn't really find any usages of such pattern: https://grep.app/search?q=def%20__post_init__%5C(self,%5B%5E%3D%5Cn%5D+%3D&regexp=true&filter%5Blang%5D%5B0%5D=Python

**Comment by AlexWaygood** [comment]
I agree this is a reasonable thing to have a rule for. There's no chance of false positives, and I think I remember hitting this footgun myself at one point in a previous project. The `RUF` category sounds good to me.
